client/pages/topsecret.jsx

```diff
@@ -1,43 +1,173 @@
 import React from 'react';
 import * as ReactBS from 'react-bootstrap';
-import {Navigation, Router} from 'react-router';
+import {Navigation} from 'react-router';
+import _ from 'lodash';
 
 const HomePage = React.createClass({
+
   mixins : [Navigation],
+
+  getInitialState() {
+    return {
+      initialBankSavings : 4294967296.69, initialExpenses : 0, tools : [],
+      addlbl : '', skuin : '', typein : '',
+      namein : '', costin : '', costQ : '', nameQ : '', searchdebug : '',
+      costfilter : '>',
+    };
+  },
+
   componentDidMount() {
     var {userid} = this.context.router.getCurrentQuery();
-    if(userid != 1)
-    {
-      //router.transitionTo('/');
+    if (userid != 1) {
+      // router.transitionTo('/');
       this.replaceWith('/');
-      console.log(this.context);
     }
 
+
   },
+  handleChange(key, event) {
+      this.setState({[key] : event.currentTarget.value});
+    },
+
+  handleAddClick(event) {
+      const doesExist = _.find(this.state.tools, { sku : this.state.skuin });
+      if (this.state.skuin == '' || this.state.namein == ''
+      || this.state.typein == '' || this.state.costin == '') {
+        this.setState({addlbl :
+         'You have left an input field blank, please fill out all input fields.'});
+        return;
+      }
+
+      this.setState({addlbl : 'The tool you enterred already exists in the armory!'});
+      if (!isNaN(this.state.costin)) {
+        if (!doesExist || this.state.tools.length == 0) {
+           // array
+          var temparray = this.state.tools.slice();
+          var newexp = Number(this.state.initialExpenses) + Number(this.state.costin);
+          temparray.push({sku : this.state.skuin,
+            name : this.state.namein, type : this.state.typein, cost : Number(this.state.costin)});
+           // update labels
+          this.setState({initialExpenses : newexp,
+             tools : temparray,
+             addlbl : 'The item with SKU: ' + this.state.skuin + ', Type: ' + this.state.typein
+              + ', Name: ' + this.state.namein + ', and Cost: $' + this.state.costin
+              + ' has been added to the armory.'},
+          );
+
+
+        } else {
+          this.setState({addlbl : 'The tool you enterred already exists in the armory!'});
+        }} else {
+        this.setState({addlbl :
+         'You have enterred an invalid cost; enter only numbers (XXXX.XX or XXXX)'});
+      } },
+
+  itemFinder(tool) {
+        let fnd, blankn, blankc, c1, c2, c3;
+        if (tool.name.indexOf(this.state.nameQ) > -1) fnd = true;
```

Comment:
This check is valid, and some developers prefer vanilla javascript, so don't worry about changing the if statement here, however the following answer is also valid using lodash.

```

    if (_.contains(tool.name, this.state.nameQ)) {
      itemNameMatch = true;
    }
```